src/pin/index.ts

```diff
@@ -68,12 +73,14 @@ export type PinQueryType = 'recursive' | 'direct' | 'indirect' | 'all'
 export interface PinLsOptions extends HTTPRPCOptions {
   paths?: CID | CID[] | string | string[]
   type?: PinQueryType
+  name?: string
 }
 
 export interface PinLsResult {
   cid: CID
   type: PinType | string
   metadata?: Record<string, any>
+  name?: string
```

Comment:
If Kubo doesn’t return this field, does it need to be in the interface?

Reply:
Kubo does return it if you call pin.ls with the `name` param. Otherwise not. So yes it needs to be there as optional.

Reply:
I think the reason it's not returned by default is that one CID can have multiple names it's pinned under.

Reply:
Aha, gotcha - so it just needs a test that shows this then all is good.

Reply:
Alright, so now we test for both cases:

- `should list all types of pins` tests that `name` isn't returned in the regular case
- `should list multiple partially matched named pins` and `should list specific named pin` tests that name is returned

Reply:
That's perfect, thanks for making those changes.